portfolio/src/main/java/com/google/sps/servlets/UserInfoServlet.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import com.google.sps.data.UserDao;
+import com.google.sps.data.UserDaoDatastore;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/user-info")
+public class UserInfoServlet extends HttpServlet {
+  private UserDaoDatastore userStorage = new UserDaoDatastore();
+
+  @Override
+  public void init() {
+    UserDaoDatastore userStorage = new UserDaoDatastore();
+  }
+
+  private String toJson(String status) {
+    Gson gson = new Gson();
+    return gson.toJson(status);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String nickname = getUserNickname(userService.getCurrentUser().getEmail());
+      response.getWriter().println(toJson(nickname));
```

Comment:
Everything looks great so far, but I see one spot for some improvement in a future PR:

I think we should return the entire User object rather than just nickname. In the future, it's reasonable to expect we'd add more fields to user's (like name, favorite book, profile pic, etc....). We could just return all the userInfo here (as JSON) and then be able to use any of the fields in the webapp.